### ISSUE TITLE
Cannot wake the device up

### ISSUE THREAD

**Issue by ngoquang2708** [ngoquang2708/android_device_samsung_vivalto3gvn#101]
Cannot wake the device up even with the power key.

**Comment by diepquynh** [comment]
Needs reverting autosleep commit from system_core, or update kernel (idk about this)
https://github.com/LineageOS/android_system_core/commit/021d60be98ca2eb72b2c63e3b1fea2169b7c2b69

**Comment by ngoquang2708** [comment]
I set the sleep time to 30min :)

**Comment by ngoquang2708** [comment]
Have you received my email?

**Comment by diepquynh** [comment]
Now I noticed this XD

**Comment by ngoquang2708** [comment]
Oh, don't read email guy :)
Anyway. Did the new rild crash?

**Comment by ngoquang2708** [comment]
Or it is running normally but you cannot make call?

**Comment by diepquynh** [comment]
rild didn't crash, at least if you don't use multisim's `RIL_onUnsolicitedResponse`, because it has 4 arguments and our libreference-ril_sp has 3 arguments, causing segfaults

**Comment by diepquynh** [comment]
Yeah I couldn't make calls. Somehow on dialing function or when it sends `AT+SSAM=0` command to modem, the RIL got crashed

**Comment by ngoquang2708** [comment]
Getting rild up and running is a big success here for you. Mine always crash. I have a fear that LD_PRELOAD cause it. I will try to shim it instead.

**Comment by diepquynh** [comment]
That was just a little bit disassembling and understanding pseudo-C codes using IDA Pro XD

What I am trying to understand is why Sammy uses such big size for its `RIL_CardStatus_v6` struct. AOSP has 280 but my stock blob has 440. Inserting ints to that struct in AOSP isn't a good idea

**Comment by ngoquang2708** [comment]
LD_PRELOAD some how cash my rild (aosp ril from your repo) :D. Now it can run with shim. So I have to modify my blobs and my shim :).

**Comment by diepquynh** [comment]
So you got signals right?

**Comment by ngoquang2708** [comment]
Nothing yet. rild just not crash.

**Comment by diepquynh** [comment]
Maybe I know why it crashes. `property_get/set` args have been modified in O
Got this crash during O bringup with my custom init binary also

**Comment by ngoquang2708** [comment]
What changes?

**Comment by diepquynh** [comment]
Oh damn forget it. I used GetProperty() for my init instead of property_get() XD

**Comment by diepquynh** [comment]
Just checked, stuffs didn't change

**Comment by ngoquang2708** [comment]
Haha. Same stuff for C, C++, Java. We all got confused. Isn't it!

**Comment by ngoquang2708** [comment]
OMG. I got signal :o

**Comment by diepquynh** [comment]
So you're surprised like me XD

**Comment by diepquynh** [comment]
I guess your RIL bugs now are exactly the same as mine

**Comment by ngoquang2708** [comment]
Yeah. No call. But USSD :D. Both SIM.
Se do you know where we should make changes in Java?
Previously we have used BOAD_RIL_CLASS to make changes. Now it seems to be gone.

**Comment by diepquynh** [comment]
That `BOARD_RIL_CLASS` is ditched in O. We all have to rebuild libril and since, most of java changes are just for that prebuilt lib so, if we were doing changes for RIL class before, do the same for libril at present

To be honest, sprd RIL is AOSP-friendly like other Samsung devices. Just because we were lazy to get it working though, or else it's very easy to bring up stuffs as for now XD
